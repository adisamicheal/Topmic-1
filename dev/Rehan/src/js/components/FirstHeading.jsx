import React from 'react';
import PropTypes from 'prop-types'

 class FirstHeading extends React.Component {
	constructor(){
		super();
	}
	render(){
		let {heading,text}=this.props;
		
		return (
			<div className="first">
				<div className="firstPart">
					<h2>{heading}</h2>
					<div className="box"></div>
					<p>{text}</p>
				</div>
				<div className="clear"></div>
			</div>
		)
	}
} 

export default FirstHeading

FirstHeading.propTypes={
	//getAction: PropTypes.func.isRequired
}
