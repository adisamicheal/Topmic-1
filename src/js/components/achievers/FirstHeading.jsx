import React from 'react';
import { string } from 'prop-types';

const FirstHeading = ({ heading, text }) => (
	<div className="first">
		<div className="firstPart">
			<div className="h2">{heading}</div>
			<div className="box" />
			<p>{text}</p>
			<div className="clear" />
		</div>
		<div className="clear" />
	</div>
);

export default FirstHeading;

FirstHeading.propTypes = {
	heading: string.isRequired,
	text: string.isRequired
};
