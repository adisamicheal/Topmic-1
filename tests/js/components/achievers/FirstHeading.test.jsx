import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FirstHeading from '../../../../src/js/components/achievers/FirstHeading'

Enzyme.configure({adapter:new Adapter()})

describe('Checks the rendering of the FirstHeading',()=>{

	const achieversData={
	firstPart:{
		heading:'',
		text:''
		
	},
	class17Data:{
		textYear:'',
		data:[
			{
				textHeading:'',
				text:'',
				image:''
			},
			{
				textHeading:'Ahmed',
				text:'',
				image:''
			},
			{
				textHeading:'Waqar',
				text:'',
				image:''
			},
			{
				textHeading:'Ahsan',
				text:'',
				image:''
			},
			{
				textHeading:'Hassan',
				text:'',
				image:''
			}
		]
	},
	class16Data:{
		textYear:'',
		data:[
			{
				textHeading:'Rehan',
				text:'',
				image:''
			},
			{
				textHeading:'Taimoor',
				text:'',
				image:''
			},
			{
				textHeading:'Mushraf',
				text:'',
				image:''
			}
		]
	}
}
	const wrapper=shallow( <FirstHeading heading={achieversData.firstPart.heading} text={achieversData.firstPart.text}/>)
	it('checks the rendering of component',()=>{
		expect(wrapper).toMatchSnapshot();
	})
})