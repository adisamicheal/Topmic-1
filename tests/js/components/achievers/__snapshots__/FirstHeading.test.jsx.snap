// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks the rendering of the FirstHeading checks the rendering of component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FirstHeading
    heading=""
    text=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="firstPart"
        >
          <h2>
            
          </h2>
          <div
            className="box"
          />
          <p>
            
          </p>
          <div
            className="clear"
          />
        </div>,
        <div
          className="clear"
        />,
      ],
      "className": "first",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              
            </h2>,
            <div
              className="box"
            />,
            <p>
              
            </p>,
            <div
              className="clear"
            />,
          ],
          "className": "firstPart",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "box",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clear",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clear",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="firstPart"
          >
            <h2>
              
            </h2>
            <div
              className="box"
            />
            <p>
              
            </p>
            <div
              className="clear"
            />
          </div>,
          <div
            className="clear"
          />,
        ],
        "className": "first",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                
              </h2>,
              <div
                className="box"
              />,
              <p>
                
              </p>,
              <div
                className="clear"
              />,
            ],
            "className": "firstPart",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "box",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clear",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
